Reqnroll/CucumberMessages/PayloadProcessing/NdjsonSerializer.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Io.Cucumber.Messages.Types;
+using Reqnroll.CucumberMessages.PayloadProcessing.Cucumber;
+using System;
+using System.IO;
+using System.Text.Json;
+
+namespace Reqnroll.CucumberMessages.PayloadProcessing
+{
+    /// <summary>
+    /// When using System.Text.Json to serialize a Cucumber Message Envelope, the following serialization options are used.
+    /// Consumers of Cucumber.Messages should use these options, or their serialization library's equivalent options.
+    /// These options should work with System.Text.Json v6 or above.
+    /// </summary>
+    public class NdjsonSerializer
```

Comment:
In https://github.com/cucumber/gherkin/pull/406, I used System.Text.Json to parse NdJson without splitting each new line explicitly.
Could we use the same approach here?
Note: This would require System.Text.Json 9.0, which I'm unsure about.

Reply:
I'll take a look at that PR.
As long as we have a consensus within the committers group that moving to .Net 9 is appropriate, then I have no issue with this.